### PR TITLE
add dac dithering control

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: hal/properties.c [hunk 1]
```diff
@@ -147,6 +147,57 @@ static int hdlr_invalid (const char* data, char* ret) {
...
+	char buf[] = "mcu -c a -l \0";
+
+	r = sscanf( data, "%d", &en );
+	if ( 1 != r ) {
+		return RETURN_ERROR;
+	}
+	if ( en < 0 || en > 1 ) {
+		return RETURN_ERROR_PARAM;
+	}
+	snprintf( buf, sizeof( buf ) - 1, "mcu -c a -l %u", en );
```
Comment (howardpang28) [inline hal/properties.c]: It doesn't seem like there's room to populate the %u in the buffer.

File: hal/properties.c [hunk 2]
```diff
@@ -147,6 +147,57 @@ static int hdlr_invalid (const char* data, char* ret) {
 	return RETURN_ERROR_SET_PROP;
 }
 
+static int hdlr_tx_a_rf_dac_dither_en (const char* data, char* ret) {
+	int r;
+	int en;
+	char buf[] = "mcu -c a -l \0";
+
+	r = sscanf( data, "%d", &en );
```
Comment (howardpang28) [inline hal/properties.c:156]: It's a bit cleaner if you just do: if (sscanf( data, "%d", &en ) != 1)
Unless you have a strong reason against it.

File: hal/properties.c (hunk 1 above)
Comment (howardpang28) [reply hal/properties.c]: Is the code tested to work?

File: hal/properties.c [hunk 3]
```diff
@@ -1679,6 +1793,63 @@ static int hdlr_rx_b_pwr (const char* data, char* ret) {
...
+	if ( db < 6 || db > 96 || 0 != db % 6 ) {
+		return RETURN_ERROR_PARAM;
+	}
+
+	sprintf( ret, "%u", db );
+
+	 // 96 dB (max) := 0x0, 90 dB := 0x1, ... 6 dB (min) := 0xf
+	sel = 16 - ( db / 6 );
+
+	snprintf( buf, sizeof( buf ), "dac -c c -b %u\r", sel );
```
Comment (cfriedtpv) [inline hal/properties.c]: Tesing inline comments

**Pull request merged**